readyplayerme/rpm-unity-sdk-core
# [SDK-553] chore: refactor and extracted shared logic from network packages

## [SDK-553](https://ready-player-me.atlassian.net/browse/SDK-553)

## Description

-   This change removes the shared logic from the networking plugins into core so it can re-used in other packages






## How to Test

-   to test you can create a new project and import these 3 packages. 
https://github.com/readyplayerme/rpm-unity-sdk-core.git#feature/extract-network-shared-logic
https://github.com/readyplayerme/rpm-unity-sdk-netcode-support.git#feature/extract-shared-logic
https://github.com/readyplayerme/rpm-unity-sdk-photon-support.git#feature/remove-shared-logic
After that is done you need to check for errors (both in editor and running the sample scenes)
Then Also try the TransferPrefab menu items. 
![image](https://github.com/readyplayerme/rpm-unity-sdk-core/assets/7085672/9961153b-6347-43cd-9ae6-ac0f0da0263f)



## Checklist








[SDK-553]: https://ready-player-me.atlassian.net/browse/SDK-553?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
